src/include/duckdb/storage/arena_allocator.hpp

```diff
@@ -83,6 +84,17 @@ class ArenaAllocator {
 		return new (mem) T(std::forward<ARGS>(args)...);
 	}
 
+	String MakeString(const char *data, const idx_t len) {
+		const auto mem = AllocateAligned(sizeof(char) * len);
+		memcpy(mem, data, len);
+
+		return String((char *)mem, len); // NOLINT
```

Comment:
If I understand the code correctly, this performs a copy of the memory to the `string owned_data` right?
And that looks wrong to me. We need to call StringView or some similar method.